### PR TITLE
Support for Sequel

### PR DESCRIPTION
**Pull request by kennym**
This is a WIP for Sequel 4 support in `stream-rails`.

Right now it just works, but needs refactoring and is missing specs.

Usage:
```
class Follow < Sequel::Model
  include StreamRails::Activity
  as_activity

  many_to_one :user
  many_to_one :favorite_user, :class => :User

  [snip]

  def after_create
    super
    add_to_feed
    StreamRails.feed_manager.follow_user(self.user.id, self.favorite_user.id)
  end

  def before_destroy
    StreamRails.feed_manager.follow_user(self.user.id, self.favorite_user.id)
    remove_from_feed
    super
  end
end
```
Current status:
* Documentation **COMPLETE**
* Specs **COMPLETE**
* Refactoring **COMPLETE**

### CODE REVIEW

File: lib/stream_rails/sync_policies.rb [hunk 1]
```diff
@@ -5,7 +5,20 @@ module SyncPolicy
     module SyncCreate
 
       def self.included(base)
-        base.after_commit :add_to_feed, on: :create
+        if base.respond_to? :after_commit
+          base.after_commit :add_to_feed, on: :create
+        else
+          base.instance_eval do
+            define_method(:add_to_feed) {
```
Comment (kennym) [inline lib/stream_rails/sync_policies.rb]: @tbarbugli any ideas on how I could avoid the redundancy. The `add_to_feed` method is already defined as a private method in the module.

Sequel doesn't seem to support `after_commit` hooks like AR. I currently have to overwrite the `after_create` and `before_destroy` hooks in Sequel and call the `add_to_feed` and `remove_from_feed` methods in there which is a huge PITA.

Any suggestions?

Comment (tbarbugli) [inline lib/stream_rails/sync_policies.rb]: not sure I get it, can you add a code snippet on how you hook save/delete callbacks with Sequel?

Comment (kennym) [inline lib/stream_rails/sync_policies.rb]: yes, there's a an example in the PR description

Comment (kennym) [inline lib/stream_rails/sync_policies.rb]: @tbarbugli ^^^

**Comment by tbarbugli** [comment]
@kennym isn't easierif you define Sequel's hooks methods (after_create/after_destroy) in sync_policies.rb directly?

**Comment by kennym** [comment]
@tbarbugli I actually started to like it this way. I can now control when an activity gets created, as it's not always desired to create an activity after the creation of an object.

Also I don't know how to overwrite the after_*, before_* methods in Sequel without overriding the methods defined in the models.

**Comment by kennym** [comment]
@tbarbugli what would have to be done to get this merged?

**Comment by tbarbugli** [comment]
@kennym best way then would be have a `activity_should_sync` method which is called when the object is created to decide if an activity should be created or not. by default that method would be implemented in `StreamRails::Activity` base class like this

```ruby
def activity_should_sync
  true
end
```

**Comment by tbarbugli** [comment]
@kennym do you need any help to finish this PR? I really like the work you've done and I think this PR is almost good to merge (docs/examples and specs are the main missing things here)

**Comment by kennym** [comment]
@tbarbugli I have found a possible way to remove the duplication: https://github.com/algolia/algoliasearch-rails/commit/d462c125e1e878a91b398c7cf8b95e607f50726e#diff-dfaec67e907148ea8211e17340f0f30cR252

...but if I do it like that, it doesn't work, yet. I will try again, but if you could take a look that would be awesome.

**Comment by tbarbugli** [comment]
nice catch!

**Comment by kennym** [comment]
> nice catch!
The funny thing is that *I* sent that pull request from the link I sent you, and I had totally forgotten about it until yesterday. Forgetting code is real. :wink:

**Comment by kennym** [comment]
@tbarbugli I figured it out. We're ready to roll.

**Comment by kennym** [comment]
@tbarbugli specs added

**Pull request merged**